lrpar/src/lib/ctbuilder.rs

```diff
@@ -892,57 +869,57 @@ pub fn _reconstitute<'a, StorageT: Deserialize<'a> + Hash + PrimInt + Unsigned>(
     (grm, sgraph, stable)
 }
 
-/// attempts to make `other` relative to `base`.
-///
-/// This is of particular note when the generated files are
-/// in different directories, or have non-trivial paths.
-///
-/// This implementation:
-/// - doesn't handle `..` components properly for input paths.
-/// - assumes both path's end with a file, not a directory.
-#[doc(hidden)]
-pub fn make_relative<A,B>(base: A, other: B) -> PathBuf
-where
-    A: AsRef<Path>,
-    B: AsRef<Path>,
-{
-    use std::{path::Component, ffi::OsStr};
-
-    let base = base.as_ref();
-    let other = other.as_ref();
-    match other.ancestors().filter(|ancestor| base.starts_with(ancestor)).next() {
-        Option::Some(minimum_shared_path) => {
-            base
-                .strip_prefix(minimum_shared_path)
-                .unwrap()
-                .components()
-                .skip(1) // skip the file name
-                .filter_map(|item| match item {
-                    Component::Normal(_) => Some(OsStr::new("..")),
-                    _ => None
-                })
-                .chain(
-                    other
-                        .strip_prefix(minimum_shared_path)
-                        .unwrap()
-                        .components()
-                        .filter_map(|item| match item {
-                            Component::Normal(x) => Some(x),
-                            _ => None
-                        })
-                )
-                .collect()
+fn serialize_bin_output<T: Serialize + ?Sized>(
+    ser: &T,
+    name: &str,
+    buffer: &mut String) -> Result<(), Box<dyn Error>> {
+    let mut w = ArrayWriter::new(name);
+    serialize_into(&mut w, ser)?;
+    let data = w.finish();
+    buffer.push_str(&data);
+    Ok(())
+}
+
+/// Makes formatting bytes into a rust array relatively painless.
+struct ArrayWriter {
+    buffer: String,
+}
+impl ArrayWriter {
+   
+    /// create a new array with the specified name
+    fn new(name: &str) -> Self {
+        Self {
+            buffer: format!(r#"#[allow(dead_code)] const {}: &[u8] = &["#, name),
         }
-        Option::None => {
-            // nothing in the paths is shared.
-            PathBuf::from(other)
+    }
+
+    /// complete the array, and return the finished string
+    fn finish(self) -> String {
+        let mut buffer = self.buffer;
+        buffer.push_str("];\n");
+        buffer
+    }
+}
+impl Write for ArrayWriter {
+    #[allow(dead_code)]
+    fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
+        use std::fmt::Write;
```

Comment:
This should probably go with the other `use`s at the top of the file?

Reply:
I don't think so. We already have `std::io::Write` within the namespace, so bringing in `std::fmt::Write` generates a collision.

Reply:
That's a good point -- I hadn't encountered that particular clash before. I *think* you can use `as` to rename the two things out of the way of each other. Whether that's worth doing or not is another thing.

Reply:
Hmm the `as` alias would work.

Reply:
Updated the PR to use it.